### DISCUSSION TITLE
IPOPT termination condition

### DISCUSSION THREAD

**Discussion by ajoardar007** [casadi/casadi#3100] [Q&A]
Hello,

While trying to solve a NLP using IPOPT (Matlab), I am unable to get the solver to terminate naturally. Upon looking through the IPOPT [docs](https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_Termination), I see that there are 4 criterias, including:

- 'tol' <1e-08
- 'dual_inf_tol' <1
- 'constr_viol_tol' <1e-04
- 'compl_inf_tol' < 1e-04

Going back to the [output](https://coin-or.github.io/Ipopt/OUTPUT.html) that IPOPT drops by default, I was trying to match as many of the above points as possible to understand why the solver has still not terminated.
Is it appropriate for me to draw the following conclusions:
- Can 'tol' (NLP error) be inferred from the 'objective' output? 
- Can 'dual_inf_tol' be inferred as the 'inf_du' output (infinity-norm (max) of the internal dual infeasibility)?
- Can 'compl_inf_tol' be inferred as the 'inf_pr' output (infinity-norm (max) of the (unscaled) constraints)?

I am getting these kinds of outputs.
iter        objective                 inf_pr          inf_du         lg(mu)       ||d||            lg(rg)   alpha_du        alpha_pr       ls
1020     9.2803977e+000    5.69e-010    4.86e-001      -5.7      5.71e-007    5.9       1.00e+000     1.00e+000h  1
1021     9.2803977e+000    2.45e-010    5.53e-001      -5.7      2.43e-007    6.4       1.00e+000     1.00e+000h  1
1022     9.2803975e+000    1.22e-009    9.39e-001      -5.7      1.24e-006    5.9       1.00e+000     1.00e+000h  1
1023     9.2803975e+000    5.47e-010    1.09e+000     -5.7      5.40e-007    6.3       1.00e+000     1.00e+000h  1

While there is hardly any change in the objective, and the constraints are well met, the solver is still not stopping. Is there some way to display the above 4 criterions at the end of every iteration?

Thanks,
Arnab

**Comment by jaeandersson** [comment]
This is a question to the IPOPT community (forum, mailing list?).
